### ISSUE TITLE
Create new table with pre-seeded column /  row names

### ISSUE THREAD

**Issue by psychemedia** [JupyterPhysSciLab/DataTable#7]
Hi 

Useful package, thanks for sharing.

This suggestion partially related to the ability to create a table from a *pandas* dataframe: create a table pre-seeded with column names, or row names, in a list. 

For example: 

`input_table.create_input_table(colnames=["col1", "col2"])`

The initial height / width of the table could be pre-seeded from such arguments.

Table sizing could also be pre-seeded by eg `input_table.create_input_table(cols=3)`

Slightly more polished might be eg a single `cols` parameter that identifies an integer or a list and behaves accordingly. 

Keeping them separate might allow you to pass in a list of colnames fewer than the number of cols, eg to allow students edit the table to add additional column names into pre-made slots for them (rather than having to add columns, for example).

**Comment by gutow** [comment]
Thanks for the suggestion.

I think most of what is necessary to implement this idea is already in the code. The `input_table.create_input_table()` call would need to be extended to accept the different keyword options. I think the default behavior would have to be kept. Before we can consider implementing it we need a complete specification for behavior.  If column or row names are provided would they be editable initially or locked/frozen? Are `ncols`, `nrows`, `columns=["",""...]`, `rows = ["",...]`, `title=".."` all that is necessary? Which would be optional and what combinations are allowed?

I am not quite clear on the last part of the suggestion about editing column names. Presently once the names are set, they cannot be edited. If I understand what is being asked for, you want the column names to be editable by the students after the fact. If so, I think that should be a separate issue.

**Comment by psychemedia** [comment]
Hi

Re: the last part of the suggestion, apols for that, it was a bit feature creep (yes, I was thinking oaf activities where students are presented with a half-finished table, with a number of columns unspecified that they would need to name themselves, which would require making them editable).

Yes, absolutely re: retaining the default behaviour. I was just thinking about situations where the column names etc may be available; rin such cases, rekeying runs the risk of introducing errors, so if names are available, it makes sense to reuse them.

**Comment by gutow** [comment]
I am unlikely to have a chance to get to this for a while. Thus here is what I think the specification should be for my future reference or someone else who wants to work on this.
* If no keywords are provided the current behavior would be maintained.
* The following keywords would be accepted:
    * `cols` accepting either an integer specifying the number of columns or a list of strings naming the columns. Required.
    * `rows` accepting either an integer specifying the number of rows or a list of strings naming the rows. Required.
    * `title` accepting a string caption or title for the table. Required.
    * `locked_labels` either True or False. If False the table would be created with the labels in the editable state. Default = True. Presently no other actions on tables are supported until the labels have been locked.

This would not support creating tables with data. That would be supported by `table_from_pandas()` (#6) `table_from_csv()`  (#5 ) methods.
